scipy/special/_comb.pyx

```diff
@@ -0,0 +1,54 @@
+from libc.math cimport log
+cdef extern from "limits.h":
+    Py_ssize_t LONG_MAX
+
+
+# Compute N choose k in integer arithmetic.
+def _comb_int(Py_ssize_t N, Py_ssize_t k):
+    cdef:
+        object numerator, denominator
+        Py_ssize_t j, M, nterms
+    
+    if (k > N) or (N < 0) or (k < 0):
+        return 0
+
+    M = N + 1
+    nterms = min(k, N - k)  
+
+    # delegate to fully typed version unless overflow
+    # The condition is rather loose:
+    #    numerator < M**nterms
+    # hence log(numerator) < nterms * log(M) < log(LONG_MAX)
+    if nterms*log(1.0*M) < log(1.0*LONG_MAX):
```

Comment:
Also, Py_ssize_t can in principle be smaller than long.
So either the "unsafe" routine should be using long, or the LONG_MAX constant be replaced with the size constant corresponding to Py_ssize_t. I'd suggest the former.